quadriga/src/main/java/edu/asu/diging/quadriga/web/DisplayCollectionController.java

```diff
@@ -0,0 +1,112 @@
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import edu.asu.diging.quadriga.core.exceptions.CollectionNotFoundException;
+import edu.asu.diging.quadriga.core.exceptions.InvalidObjectIdException;
+import edu.asu.diging.quadriga.core.model.Collection;
+import edu.asu.diging.quadriga.core.model.EventGraph;
+import edu.asu.diging.quadriga.core.model.MappedTripleGroup;
+import edu.asu.diging.quadriga.core.model.MappedTripleType;
+import edu.asu.diging.quadriga.core.model.mapped.Predicate;
+import edu.asu.diging.quadriga.core.service.CollectionManager;
+import edu.asu.diging.quadriga.core.service.EventGraphService;
+import edu.asu.diging.quadriga.core.service.MappedTripleGroupService;
+import edu.asu.diging.quadriga.core.service.PredicateManager;
+
+@Controller
+public class DisplayCollectionController {
+
+    @Autowired
+    private CollectionManager collectionManager;
+
+    @Autowired
+    private EventGraphService eventGraphService;
+    
+    @Autowired
+    private MappedTripleGroupService mappedTripleGroupService;
+    
+    @Autowired
+    private PredicateManager predicateManager;
+    
+    private Logger logger = LoggerFactory.getLogger(getClass());
+
+    @RequestMapping(value = "/auth/collections/{id}", method = RequestMethod.GET)
+    public String get(@PathVariable String id, Model model) {
+        
+        Collection collection;
+        try {
+            collection = collectionManager.findCollection(id);
+            if(collection == null) {
+                return "error404Page";
+            }
+        } catch (InvalidObjectIdException e) {
+            return "error404Page";
+        }
+        
+        ObjectId collectionId = collection.getId();
+        
+        List<EventGraph> eventGraphs = eventGraphService.findAllEventGraphsByCollectionId(collectionId);
+        
+        if(!eventGraphs.isEmpty()) {
+            // We get last network submission info by getting the last EventGraph which will be a part of the last network
+            EventGraph lastNetwork = eventGraphs.get(0);
+            model.addAttribute("lastNetworkSubmittedAt", lastNetwork.getCreationTime().atZoneSameInstant(ZoneId.systemDefault()));
+            model.addAttribute("lastNetworkSubmittedBy", lastNetwork.getSubmittingApp());
+        }
+        
+        model.addAttribute("collectionName", collection.getName());
+        model.addAttribute("description", collection.getDescription());
+        model.addAttribute("creationTime", collection.getCreationTime().atZoneSameInstant(ZoneId.systemDefault()));
+        
+        // One network may have multiple eventGraphs, but all of them will have same sourceURI in the context
+        // This sourceURI will be used to group eventGraphs together that belong to the same network
+        model.addAttribute("numberOfSubmittedNetworks", eventGraphs.stream().collect(Collectors.groupingBy(event -> event.getContext().getSourceUri())).size());
```

Comment:
this doesn't seem like a good approach. What if there are 10000 or more graphs in the db. This be done via MongoDb aggregations.